# Gun Violence Data Cleaning and Consistency Checks

In [2]:
# Import Libraries
import pandas as pd
import numpy as np
import os

In [3]:
# path to data folde
path = r'C:\Users\justi\OneDrive\Desktop\Class work\Achievement 6\Data'

In [4]:
#import gun violence data set
gv = pd.read_csv(os.path.join(path, 'original data', 'gv data.csv'), index_col = False)

In [5]:
gv.shape

(239677, 8)

In [6]:
gv.head()

incident_id      date           state city_or_county  n_killed  n_injured  \
0       461105  1/1/2013    Pennsylvania     Mckeesport         0          4   
1       460726  1/1/2013      California      Hawthorne         1          3   
2       478855  1/1/2013            Ohio         Lorain         1          3   
3       478925  1/5/2013        Colorado         Aurora         4          0   
4       478959  1/7/2013  North Carolina     Greensboro         2          2   

   latitude  longitude  
0   40.3467   -79.8559  
1   33.9090  -118.3330  
2   41.4455   -82.1377  
3   39.6518  -104.8020  
4   36.1140   -79.9569

In [7]:
gv.describe()

incident_id       n_killed      n_injured       latitude  \
count  2.396770e+05  239677.000000  239677.000000  231754.000000   
mean   5.593343e+05       0.252290       0.494007      37.546598   
std    2.931287e+05       0.521779       0.729952       5.130763   
min    9.211400e+04       0.000000       0.000000      19.111400   
25%    3.085450e+05       0.000000       0.000000      33.903400   
50%    5.435870e+05       0.000000       0.000000      38.570600   
75%    8.172280e+05       0.000000       1.000000      41.437375   
max    1.083472e+06      50.000000      53.000000      71.336800   

           longitude  
count  231754.000000  
mean      -89.338348  
std        14.359546  
min      -171.429000  
25%       -94.158725  
50%       -86.249600  
75%       -80.048625  
max        97.433100

In [8]:
# Checking for NAN records
gv.isnull().sum()

incident_id          0
date                 0
state                0
city_or_county       0
n_killed             0
n_injured            0
latitude          7923
longitude         7923
dtype: int64

# There are 7923 missing records in the latitude and longitude columns 

## I will not be doing anything with the records in the missing latitude/longitude columns as they only account for 3% of the data and there arent any missing values in the city/state columns

In [9]:
#Check
gv.head()

incident_id      date           state city_or_county  n_killed  n_injured  \
0       461105  1/1/2013    Pennsylvania     Mckeesport         0          4   
1       460726  1/1/2013      California      Hawthorne         1          3   
2       478855  1/1/2013            Ohio         Lorain         1          3   
3       478925  1/5/2013        Colorado         Aurora         4          0   
4       478959  1/7/2013  North Carolina     Greensboro         2          2   

   latitude  longitude  
0   40.3467   -79.8559  
1   33.9090  -118.3330  
2   41.4455   -82.1377  
3   39.6518  -104.8020  
4   36.1140   -79.9569

In [10]:
# Duplicates
gv_dups = gv[gv.duplicated()]

In [11]:
gv_dups

Empty DataFrame
Columns: [incident_id, date, state, city_or_county, n_killed, n_injured, latitude, longitude]
Index: []

# There are no duplicates

In [15]:
gv.rename(columns = {'n_killed' : 'number_killed', 'n_injured' : 'number_injured'}, inplace = True)

In [17]:
#Check
gv.head()

incident_id      date           state city_or_county  number_killed  \
0       461105  1/1/2013    Pennsylvania     Mckeesport              0   
1       460726  1/1/2013      California      Hawthorne              1   
2       478855  1/1/2013            Ohio         Lorain              1   
3       478925  1/5/2013        Colorado         Aurora              4   
4       478959  1/7/2013  North Carolina     Greensboro              2   

   number_injured  latitude  longitude  
0               4   40.3467   -79.8559  
1               3   33.9090  -118.3330  
2               3   41.4455   -82.1377  
3               0   39.6518  -104.8020  
4               2   36.1140   -79.9569

In [18]:
gv.describe()

incident_id  number_killed  number_injured       latitude  \
count  2.396770e+05  239677.000000   239677.000000  231754.000000   
mean   5.593343e+05       0.252290        0.494007      37.546598   
std    2.931287e+05       0.521779        0.729952       5.130763   
min    9.211400e+04       0.000000        0.000000      19.111400   
25%    3.085450e+05       0.000000        0.000000      33.903400   
50%    5.435870e+05       0.000000        0.000000      38.570600   
75%    8.172280e+05       0.000000        1.000000      41.437375   
max    1.083472e+06      50.000000       53.000000      71.336800   

           longitude  
count  231754.000000  
mean      -89.338348  
std        14.359546  
min      -171.429000  
25%       -94.158725  
50%       -86.249600  
75%       -80.048625  
max        97.433100

In [19]:
gv.dtypes

incident_id         int64
date               object
state              object
city_or_county     object
number_killed       int64
number_injured      int64
latitude          float64
longitude         float64
dtype: object

In [20]:
# Export Clean Gun violence data
gv.to_csv(os.path.join(path, 'prepared data', 'Gun Violence Clean.csv'))